### PR TITLE
PohRecorder take Arc not &Arc for blockstore

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Takes a reference when it should be taking ownership the Arc; made it unclear in #31232 if I needed to keep the original blockstore around in the test frame.

Ongoing quest of correcting `&Arc` -> `Arc` as I find them 🔍 

#### Summary of Changes
* Take `Arc<Blockstore>` instead of `&Arc<Blockstore>` + cloning.
* Bunch of use-cases needed to be changed.

Fixes #

### CODE REVIEW

File: poh/src/poh_recorder.rs [hunk 1]
```diff
@@ -909,7 +909,7 @@ impl PohRecorder {
         next_leader_slot: Option<(Slot, Slot)>,
         ticks_per_slot: u64,
         id: &Pubkey,
-        blockstore: &Arc<Blockstore>,
+        blockstore: Arc<Blockstore>,
```
Comment (apfitzge) [inline poh/src/poh_recorder.rs:912]: Signature changed: taked owned `Arc`

File: poh/src/poh_recorder.rs [hunk 2]
```diff
@@ -980,7 +980,7 @@ impl PohRecorder {
...
```
Comment (apfitzge) [inline poh/src/poh_recorder.rs:983]: Signature changed: taked owned `Arc`

File: poh/src/poh_recorder.rs [hunk 3]
```diff
@@ -1030,7 +1030,7 @@ impl PohRecorder {
 
 pub fn create_test_recorder(
     bank: &Arc<Bank>,
-    blockstore: &Arc<Blockstore>,
+    blockstore: Arc<Blockstore>,
```
Comment (apfitzge) [inline poh/src/poh_recorder.rs:1033]: Signature changed: taked owned `Arc`

**Comment by apfitzge** [comment]
Made comments at the places where the function signatures were updated.

**Review by steviez — APPROVED** [review]
Looks good to me as well, thanks for the cleanup!

**Comment by apfitzge** [comment]
> Seems straight-forward to me. Assuming there's no weird/special cases you'd like me to inspect more closely, I'd say lgtm!

No special cases I found - just found myself a bit confused on if I needed to keep the original blockstore around after passing it to the PohRecorder creation.

**Pull request merged**